articles/bottom-navigation-bar-in-android/index.md

```diff
@@ -0,0 +1,290 @@
+* **_Name_** - name of the icon.
+* **_Size_**- the dimensions of the clipart
+* **_Color_** - provides color options for the clipart.
+* **_Opacity_** - deals with the visibility of the icon.
+
+Select on the icon next to clipart. A list of different cliparts is displayed.
+
+![Clipart List](/engineering-education/bottom-navigation-bar-in-android/list.jpg)
+
+One can search for a clipart in the search bar and choose whether the design of the clipart should be filled or outlined. We will use the outlined design.
+
+![Favorites](/engineering-education/bottom-navigation-bar-in-android/favorites.jpg)
+
+Once a clipart is selected, click *OK* then *Next* and finally *Finish*. Open the menu file in `res/menu/ bottom_navigation_menu` and add the following lines of code:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:id="@+id/navigation_home"
+        android:icon="@drawable/ic_outline_home_24"
+        android:title="@string/title_home" />
+    <item
+        android:id="@+id/navigation_search"
+        android:icon="@drawable/ic_outline_search_24"
+        android:title="@string/title_search" />
+    <item
+        android:id="@+id/navigation_favourites"
+        android:icon="@drawable/ic_baseline_favorite_border_24"
+        android:title="@string/title_favourites" />
+    <item
+        android:id="@+id/navigation_profile"
+        android:icon="@drawable/ic_baseline_perm_identity_24"
+        android:title="@string/title_profile" />
+
+</menu>
+
+```
+The **menu** file serves as a container for menu items. **Item** represents one item on the menu. Each item has a title and an icon.
+
+#### Step 6 - Let's Code.
+In our  `MainActivity.java` add the following lines of code:
+
+```java
+//Initialize Bottom Navigation View
+
+BottomNavigationView navView = findViewById(R.id.bottomNav_view);
+```
+We will initialize the Bottom Navigation view in the Activity's `onCreate` method. Next, we will initialize an `AppBarConfiguration` object. It is used to manage the behavior of the icons in the Navigation View.
+
+```java
+AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder
+```
+We will pass the ID of the different destinations in the Bottom Navigation View.
+
+```java
+  R.id.navigation_home, R.id.navigation_favourites, R.id.navigation_profile, R.id.navigation_search )
+        .build();
+```
+
+Then set up the NavController and link it with the Bottom Navigation View.
+
+```java
+//Initialize NavController.
+NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
+NavigationUI.setupWithNavController(navView, navController);
+```
+
+We are done! Let’s run the app.
+
+
+![Home](/engineering-education/bottom-navigation-bar-in-android/home.jpg)
```

Comment:
You can find a way to make a short gif to show the progress through each navigation view. And to reduce the number of Images